en/docs/samples/device-based-adaptive-auth.md

```diff
@@ -0,0 +1,111 @@
+
+2.  Click **Edit** on the `saml2-web-app-pickup-dispatch.com` service provider.
+
+3.  Expand the **Local and Outbound Configuration** section and click **Advanced Authentication**.
+
+4.  Expand **Script Based Conditional Authentication**.
+
+5.  Click **Templates** on the right side of the **Script Based Conditional Authentication** field and then click **New-Device-Based**. 
+
+    ![device-based-template](../assets/img/samples/device-based-template.png)
+
+6.  Click **Ok**. The authentication script and authentication steps
+    are configured. 
+    
+    The authentication script prompts the second step of authentication if the user logs in using a new device or browser that they have not used before. The new device or browser is identified by a cookie.
+    
+7. Set the `deviceRememberPeriod` to specify how long the device should be considered as a trusted device. 
+
+    Once this time period has elapsed, the device is considered as a new device. The time period is set to two years by default.
+    
+    ``` java
+        var deviceRememberPeriod = 60 * 60 * 24 * 365 * 2; 
+    ```
+
+8. The authentication steps added are `totp` and `fido`. However, these are authentication steps that you would normally use in production. 
+
+    To try out sample authenticators with the sample application, delete the two
+    authenticators and add the following sample authenticators instead.
+
+    1.  Click **Delete** to remove the `            totp           ` and
+        `            fido           ` authenticators from Step 2 (the
+        second authentication step).
+        
+        ![delete authenticators](../assets/img/samples/delete-authenticators.png)
+        
+    2.  Select **Demo Hardware Key Authenticator** and click **Add**.  
+        ![add new authenticator](../assets/img/samples/add-new-authenticator.png)
+
+9.  Click **Update**.
+
+----
+
+## Add a user
+
+
+1.  Start the server and log in to the [management console](insertlink).
+
+2.  Create a new user named 'Kim' with login permission. Do not assign any roles.
+
+    For instructions, see [Adding Users and Roles](insertlink).
+
+----
+
+## Try it out
+
+1.  Access the following sample PickUp application URL:
+    
+    <http://localhost.com:8080/saml2-web-app-pickup-dispatch.com>
+
+    ![dispatch-login](../assets/img/samples/dispatch-login.png)
+
+2.  Click **Login** and enter Kim's credentials. 
+
+    Note that you are prompted for hardware key authentication because Kim has never logged in from this device or browser before.  
+
+3.  Enter the 4-digit key and click **Sign-In**. 
+
+    You are successfully logged in to the application.  
+
+    ![hardware-key-authenticator](../assets/img/samples/hardware-key-authenticator.png) 
+
+4.  Log out and log in again using Kim's credentials on the same browser window. 
+
+    Note that you are logged in successfully without being prompted for hardware key authentication again. This is because this
```

Comment:
Please make sure the alignment here is correct. Looks like there are extra spaces.